import {Breadcrumb, Typography} from 'antd';
import React from 'react';
import {contentStyle} from "../SiderStaticMenu.js";
import Image from "next/image";


export default function ManualMapping() {

    return (
        <div className="modal-body-stw" style={contentStyle}>
            <Breadcrumb>
                <Breadcrumb.Item>Help</Breadcrumb.Item>
                <Breadcrumb.Item>Manual</Breadcrumb.Item>
                <Breadcrumb.Item>Spatial Mapping</Breadcrumb.Item>
            </Breadcrumb>
            <Typography style={{marginTop:50}}>
                <h1>Spatial Mapping</h1>
                <h2>1. Introduction</h2>
                <p style={{fontSize:16}}> This Module allows users to utilize multiple machine learning methods to map cell labels
                    in scRNA-seq data back to the spatial transcriptome to see the distribution of cell labels in the spatial transcriptome data.
                    It presents the predicted spatial distribution of cell labels by scRNA-seq came from users, which may provide a new
                    dimensional perspective to explore the scRNA-seq datasets.
                </p>
                <h2>2. Reference Data</h2>
                <h2>3. Methods</h2>
                <h4>(1) Tangram </h4>
                <p>Tangram, an algorithm that uses sc/snRNA-seq data as &apos;puzzle pieces&apos; to align in space to match
                    &apos;the shape&apos; of the spatial data (Fig. 1a). The input to Tangram is sc/snRNA-seq data along with
                    spatial profiling data from the same region or tissue type, from any currently available spatial
                    method (for example MERFISH, smFISH, STARmap, ISH, or Visium), requiring only that the two modalities
                    share at least some subset of common genes.</p>
                <p>
                    <i>- Biancalani, T., Scalia, G., Buffoni, L. et al. Deep learning and alignment of spatially resolved
                    single-cell transcriptomes with Tangram. Nat Methods 18, 1352â€“1362 (2021). </i>
                    <a href={"https://doi.org/10.1038/s41592-021-01264-7"} target={"_blank"} rel={"noreferrer"}>
                    https://doi.org/10.1038/s41592-021-01264-7</a>
                </p>
                <h4>(2) Cell2location </h4>
                <h2>4. Format of Uploaded Files</h2>
                <a id={"format_uploaded_files"} style={{position: 'relative', top: "-150px"}}></a>
                <h4>(1) Counts Matrix File </h4>
                <span>
                    This file contains gene expression <b>(raw counts)</b> values in which <b>columns are genes </b>
                    presented with gene names identifier (HGNC symbol name) and
                    <b> rows are cells</b> presented with cell IDs. Formats accepted are .csv, .tsv and .txt
                    in .gz/zip compression. <b>File .csv.gz with symbol name is recommended</b>.
                </span><br/>
                <span>&gt; Example:</span><br/>
                <Image src={`/images/counts_matrix_example.png`} alt="..." width={400} height={220}/>
                <h4>(2) Labels File </h4>
                <span>
                    This file is generated by users after you have annotated each cluster identified in scRNA-seq.
                    The file contains <b>two columns</b>: unique<b> &quot;cell_id&quot; </b> must in first column,
                    indicating the name of the cell; and <b>&quot;cell_type&quot;</b>, indicating the name of the
                    cluster considered. Formats accepted are .csv, .tsv and .txt in .gz/zip compression,
                    but <b>.csv.gz is recommended</b>.
                </span><br/>
                <span>&gt; Example:</span><br/>
                <Image src={`/images/labels_example.png`} alt="..." width={400} height={130}/>
                <h2>5. Mapping Result</h2>
                <a id={"mapping_result"} style={{position: 'relative', top: "-150px"}}></a>
                <h4>(1) Preprocessing </h4>
                <a id={"preprocessing"} style={{position: 'relative', top: "-150px"}}></a>
                <h4>(2) Filtering </h4>
                <a id={"filtering"} style={{position: 'relative', top: "-150px"}}></a>
                <h4>(3) Spatial Niche </h4>
                <a id={"spatial_niche"} style={{position: 'relative', top: "-150px"}}></a>
                <h4>(4) Cell Types Colocalization </h4>
                <a id={"colocalization"} style={{position: 'relative', top: "-150px"}}></a>
                <h4>(5) Cell-Cell Interactions </h4>
                <a id={"interaction"} style={{position: 'relative', top: "-150px"}}></a>
                <h4>(6) Result Files Download </h4>
                <a id={"download"} style={{position: 'relative', top: "-150px"}}></a>

            </Typography>
        </div>
    )
}